ligand/views.py

```diff
@@ -377,12 +1163,79 @@ def TargetDetails(mode, request, **kwargs):
+                                    functional_dict['class'] = protein.get_protein_class_from_slug(short=True)
+                                    functional_dict['family'] = protein.get_protein_family_from_slug(short=True)
+                                ligand_data_potency.append(functional_dict)
+
+
+
+    return(ligand_data_affinity,ligand_data_potency)
+
+def TargetDetails(mode, request, **kwargs):
+    cache_key = False
+    simple_selection = request.session.get('selection', False)
+    selection = Selection()
+    if simple_selection:
+        selection.importer(simple_selection)
+    if mode == 'extended':
+        if 'slug' in kwargs:
+            cache_key = "lig_ext_slug_" + kwargs['slug']
+            ps = AssayExperiment.objects.filter(protein__family__slug=kwargs['slug']).prefetch_related('protein','ligand', 'publication')
+
+            # Set selection for purchasability page
+            prot_ids = list(AssayExperiment.objects.filter(protein__family__slug=kwargs['slug']).values_list("protein_id", flat = True).distinct())
+            proteins = Protein.objects.filter(pk__in=prot_ids)
+            for prot in proteins:
+                selection.add('targets', 'protein', SelectionItem('protein', prot))
+            simple_selection = selection.exporter()
+            request.session['selection'] = simple_selection
+        else:
+            if selection.reference != []:
+                prot_id = [x.item for x in selection.reference]
+                if len(prot_id) > 0:
+                    cache_key = "lig_ext_protid_" + ",".join(prot_id)
+                    ps = AssayExperiment.objects.filter(protein__in=prot_id).prefetch_related('protein','ligand', 'publication').order_by('ligand__id','assay_type','value_type','source')
+
+        # if queryset is empty redirect to ligand browser
+        if not ps and 'slug' not in kwargs:
+            return redirect("ligand_selection")
+
+        if cache_key != False and cache.has_key(cache_key):
+            result = cache.get(cache_key)
+            ligand_data_affinity = result['affinity_data']
+            ligand_data_potency = result['potency_data']
+        else:
+            ligand_data_affinity,ligand_data_potency = LigandListDetails(mode, ps)
+        context = {}
+        context['affinity_data'] = ligand_data_affinity
+        context['potency_data'] = ligand_data_potency
+        cache.set(cache_key, context, 60*60*24*7)
+
+    elif mode == 'compact':
+
+        if selection.reference != []:
+            prot_id = [x.item for x in selection.reference]
+            cache_key = "lig_compact_protid_" + ",".join(prot_id)
+            ps = AssayExperiment.objects.filter(protein__in=prot_id).prefetch_related('protein', 'ligand', 'ligand__ligand_type')
+                                                                                      # 'ligand__ids__web_resource',                                                                      # 'ligand')
```

Comment:
same question as before, are these not needed anymore to be prefetched?

Reply:
Solved in [39bd4a8](https://github.com/protwis/protwis/pull/550/commits/39bd4a8230b189165eddc18d6beb1c8dbe956bd1)